packages/core/src/actors/promise.ts

```diff
@@ -42,6 +54,12 @@ export function fromPromise<T>(
       }
     },
     start: (state, { self }) => {
+      // TODO: determine how to allow customizing this so that promises
+      // can be restarted if necessary
+      if (state.status !== 'active') {
```

Comment:
the same should be added to both observable types (+tests)

Reply:
https://github.com/statelyai/xstate/pull/3743/commits/a08743226d1c009ca922dc684fb44b44d6af87dc